feature-generation/unsupervised-models/script.whizzml

```diff
@@ -0,0 +1,39 @@
+
+(define (set-params dataset params exclude)
+  (if exclude
+    (let (excluded (params "excluded_fields" [])
+          objective-id (dataset-get-objective-id dataset))
+      (if objective-id
+          (assoc params "excluded_fields" (append excluded objective-id))
+          params))
+    params))
+
+(define (try-create-model type dataset params exclude)
+  (let (final-params (set-params dataset params exclude))
+    (try
+     (log-info " - Creating " type)
+     (create type dataset final-params)
+     (catch e
+       (log-warn "Could not create the " type ": " e)))))
+
+(define cluster-id
+  (try-create-model "cluster" train-dataset cluster-params exclude-objective))
+
+(define association-id
+  (let (objective-id (dataset-get-objective-id train-dataset)
+        params (if (not (empty? association-params))
+                 association-params
+                 {"rhs_predicate" [{"field" objective-id}]}))
+    (try-create-model "association" train-dataset params false)))
```

Comment:
Does this work? So if I exclude the objective id in exclude_fields then I can ask for {"rhs_predicate" [{"field" objective-id}]}))?

Reply:
oh, sorry, you don't exclude objective here